src/analyses/ai.h

```diff
@@ -252,7 +252,8 @@ class ai_baset
 
     locationt n = std::next(l);
 
-    auto step_return = p->step(n, *(storage->abstract_traces_before(n)));
+    auto step_return =
+      p->step(n, *(storage->abstract_traces_before(n)), nullptr);
```

Comment:
Does the nullptr here imply we're conflating some states that we could potentially distinguish?

Reply:
Good question but in this case no.  The step is local so the history is not needed.  I have added a comment to this effect.

( There is a separate conversation about whether such an interface is even meaningful w.r.t. function calls and goto instructions. )